libbeat/processors/actions/alterFieldProcessor.go

```diff
@@ -133,3 +149,27 @@ func (a *alterFieldProcessor) alter(event *beat.Event, field string) error {
 
 	return nil
 }
+
+func (a *alterFieldProcessor) alterValue(event *beat.Event, valueKey string) error {
+	value, err := event.GetValue(valueKey)
+	if err != nil {
+		return fmt.Errorf("could not fetch value for key: %s, Error: %w", valueKey, err)
+	}
+
+	if v, ok := value.(string); ok {
+		err = event.Delete(valueKey)
+		if err != nil {
+			return fmt.Errorf("could not delete key: %s,  %w", v, err)
+		}
+
+		v, _ = a.alterFunc(v)
```

Comment:
I think we should check errors here, and for the `alterField` function as well. 

For now it's just `strings.ToLower` but perhaps we would add more processors in the future and they would have an error case. We risk to miss these errors if this implementation is not updated accordingly.

Reply:
for `alterField` - and I think this piece of code handles it so should be good https://github.com/khushijain21/elastic-agent-libs/blob/main/mapstr/mapstr.go#L216